kafka-exporter/src/main/java/io/opentelemetry/contrib/kafka/KafkaSpanExporter.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.kafka;
+
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import javax.annotation.Nonnull;
+import javax.annotation.concurrent.ThreadSafe;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.KafkaException;
+import org.apache.kafka.common.serialization.Serializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@ThreadSafe
+@SuppressWarnings("FutureReturnValueIgnored")
+public class KafkaSpanExporter implements SpanExporter {
+  private static final Logger logger = LoggerFactory.getLogger(KafkaSpanExporter.class);
+  private static final long DEFAULT_TIMEOUT_IN_SECONDS = 5L;
+  private final String topicName;
+  private final Producer<String, Collection<SpanData>> producer;
+  private final ExecutorService executorService;
+  private final long timeoutInSeconds;
+  private final AtomicBoolean isShutdown = new AtomicBoolean();
+
+  public KafkaSpanExporter(
+      String topicName,
+      Producer<String, Collection<SpanData>> producer,
+      ExecutorService executorService,
+      long timeoutInSeconds) {
+    this.topicName = topicName;
+    this.producer = producer;
+    this.executorService = executorService;
+    this.timeoutInSeconds = timeoutInSeconds;
+  }
+
+  public KafkaSpanExporter(
```

Comment:
Instead of adding more constructors you might want to opt in for using `static factory methods` or a builder like pattern. I would think static factory methods make the code much clearer. You could use some default names like `create` to replace the main constructor, and I've used things like basic, and with timeout etc etc as other options.

This gives the replacements names and makes it easier to work with.
But looking at all of the options you have a Builder might be a good option as well. probably with some default/required fields to start with

Reply:
A builder is added.